mods/ra/maps/red-zone/red-zone.lua

```diff
@@ -0,0 +1,608 @@
+	Trigger.AfterDelay(DateTime.Seconds(2), function()
+		Media.PlaySpeechNotification(ussr, "AlliedForcesFallen")
+		ussr.MarkCompletedObjective(EscortObjective)
+		ussr.MarkCompletedObjective(NukeObjective)
+		england_main.MarkFailedObjective(AlliedObjective)
+	end)
+end
+
+RunInitialActivities = function()
+	
+	APCDropoff()
+
+	Camera.Position = APCDropoffPoint.CenterPosition
+
+	ActivatePatrolDestroyers()
+	ActivatePatrolInfantry()
+
+	Trigger.OnCapture(Tank1, EnglandAirBaseReveal)
+	Trigger.OnCapture(Tank2, EnglandAirBaseReveal)
+
+	-- Trigger.OnPassengerEntered(Hind1, Hind1PassengerCheck)
+	Trigger.OnCapture(Hind1, Hind1PassengerCheck)
+	Trigger.OnCapture(Hind2, Hind2PassengerCheck)
+	Trigger.OnCapture(Hind3, Hind3PassengerCheck)
+
+	Trigger.OnCapture(Hind1, NuclearTimerStarted)
+	Trigger.OnCapture(Hind2, NuclearTimerStarted)
+	Trigger.OnCapture(Hind3, NuclearTimerStarted)
+
+	Trigger.OnKilled(Hind1, BackupHeli)
+	Trigger.OnKilled(Hind2, BackupHeli)
+	Trigger.OnKilled(Hind3, BackupHeli)
+
+	if Difficulty == easy or Difficulty == hard then
+		Hind1.GrantCondition(Difficulty)
+		Hind2.GrantCondition(Difficulty)
+		Hind3.GrantCondition(Difficulty)
+	end
+
+	MissionObjectivesMain = { Prison, CommunicationsCenter }
+	Trigger.OnAllKilled(MissionObjectivesMain, DemoTruckOnApproach) 
+
+	HindHelicoptersTeam = { Hind1, Hind2, Hind3 }
+	Trigger.OnAllKilled(HindHelicoptersTeam, MissionFailed)
+
+	Trigger.OnKilled(Prison, FreeVolkov)
+	Trigger.OnKilled(CommunicationsCenter, CommunicationsCenterDestroyed)
+end
+
+PlayerIsTeamAi = function(player)
+	return player.InternalName == AlphaTeamPlayerName or player.InternalName == BravoTeamPlayerName
+end
+
+PlayerIsHumanOrBot = function(player)
+	return player.IsNonCombatant == false and PlayerIsTeamAi(player) == false
+end
+
+
+WorldLoaded = function()
+	local teamPlayers = Player.GetPlayers(function(p)
+		return PlayerIsHumanOrBot(p)
+	end)
+
+	ussr = Player.GetPlayer("USSR")
+	bad_guy = Player.GetPlayer("BadGuy")
+	forEnglandJames = Player.GetPlayer("England Missile Silos")
+	neutral = Player.GetPlayer("Neutral")
+
+	england_air = Player.GetPlayer("England Air Base")
+	england_north = Player.GetPlayer("England Northern Base")
+	england_main = Player.GetPlayer("England Main Base")
+	england_prison = Player.GetPlayer("England Prison Base")
+	england_nuke = Player.GetPlayer("England Missile Silos")
+
+	Utils.Do(teamPlayers, function(p)
+		
+		PlayerInfo[p.InternalName] =
+			{
+				IsPilot = false, 
+			}
+
+		if p.IsLocalPlayer then	LocalPlayerInfo = PlayerInfo[p.InternalName] end
+	end)
+
+	RunInitialActivities()
+
+	SetupFactories()
+	SetupMainFactories()
+
+	Trigger.OnObjectiveAdded(ussr, function(p, id)
+		Trigger.AfterDelay(DateTime.Seconds(5), function()
+			Media.DisplayMessage(p.GetObjectiveDescription(id), "Primary Objective")
+		end)
+	end)
+
+	AlliedObjective = england_main.AddPrimaryObjective("Don't let soviets to destroy the Missile Silos.", england_main.Color)
```

Comment:
Use `AddObjective` instead. We are phasing out `AddPrimaryObjective` and `AddSecondaryObjective`.